Add option to disable ipv6 protocol in microcloud init command.

**Problem description:**

**microcloud init** require IPv4 and IPv6 stack.
microcloud use calls _[mdns.DefaultParams(service)](https://github.com/canonical/microcloud/blob/da6d8a4962423d0d5516b3c329c584df7761c937/microcloud/mdns/lookup.go#L147)_ which instructs to use ipv4 and ipv6 sockets in msdn Query function call.
**microcloud init** fails with message **Error: Failed lookup: write udp6 [::]:<SRC_PORT>->[ff02::fb]:5353: sendto: cannot assign requested address** in case if ipv6 is disabled.

**Prerequisites**

- The single instance with the single network interface is enough for reproduction
- install microcloud via snap: _snap install microcloud_  (neverminded which channel is used, all are impacted)
- disable ipv6 stack (for example using shell command: _sudo sysctl -w net.ipv6.conf.all.disable_ipv6=1; sudo sysctl -w net.ipv6.conf.default.disable_ipv6=1_)

**Steps to reproduce:**

- run _sudo microcloud init --auto_
execution stops after **Scanning for eligible servers ...** step with the message **Error: Failed lookup: write udp6 [::]:<SRC_PORT>->[ff02::fb]:5353: sendto: cannot assign requested address**

**Recommendations to fix**
Add option to command disable ipv6 protocol in **microcloud init** command.

Hi @tregubovav-dev, thanks for reporting this. When disabling IPv6 using `sysctl` this is causing a panic on `latest/edge`:

```
root@mc:~# sysctl -w net.ipv6.conf.all.disable_ipv6=1; sudo sysctl -w net.ipv6.conf.default.disable_ipv6=1
root@mc:~# microcloud init --auto
Waiting for LXD to start...
Using address "10.146.93.183" for MicroCloud
Scanning for eligible servers ...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5d77ae]

goroutine 1 [running]:
net.(*UDPConn).SyscallConn(0xc0001406c0?)
	net/udpsock.go:129 +0xe
golang.org/x/net/internal/socket.NewConn({0xf6f410?, 0x0})
	golang.org/x/net@v0.15.0/internal/socket/rawconn.go:57 +0x142
golang.org/x/net/ipv6.NewPacketConn({0xf6e758?, 0x0})
	golang.org/x/net@v0.15.0/ipv6/endpoint.go:121 +0x3d
github.com/hashicorp/mdns.(*client).setInterface(0xc000192900, 0xc000140768?)
	github.com/hashicorp/mdns@v1.0.5/client.go:222 +0xc5
github.com/hashicorp/mdns.Query(0xc000140808)
	github.com/hashicorp/mdns@v1.0.5/client.go:76 +0x8a
github.com/canonical/microcloud/microcloud/mdns.Lookup({0xf6b5a0, 0x155bd40}, 0xc000087da0, {0xc000191680, 0xd}, 0xc0001408a8?)
	github.com/canonical/microcloud/microcloud/mdns/lookup.go:151 +0x212
github.com/canonical/microcloud/microcloud/mdns.LookupPeers({0xf6b5a0, 0x155bd40}, 0xc000140bb8?, {0xe0fde8, 0x3}, {0xc00019160e, 0x2})
	github.com/canonical/microcloud/microcloud/mdns/lookup.go:64 +0x1b8
main.lookupPeers(0xc000087ec0, 0x1, 0x2?, 0xc0001879e0, {0x0, 0x0, 0xd?}, 0x7ffdace5b42c?)
	github.com/canonical/microcloud/microcloud/cmd/microcloud/main_init.go:217 +0x4bc
main.(*cmdInit).RunInteractive(0xc000145f80, 0x0?, {0x0?, 0x0?, 0x0?})
	github.com/canonical/microcloud/microcloud/cmd/microcloud/main_init.go:129 +0x49e
main.(*cmdInit).Run(0x0?, 0x0?, {0xc0001863f0?, 0x0?, 0x0?})
	github.com/canonical/microcloud/microcloud/cmd/microcloud/main_init.go:78 +0x36
github.com/spf13/cobra.(*Command).execute(0xc000004600, {0xc000085e40, 0x3, 0x4})
	github.com/spf13/cobra@v1.7.0/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004300)
	github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.7.0/command.go:992
main.main()
	github.com/canonical/microcloud/microcloud/cmd/microcloud/main.go:101 +0x8aa

```

--- a/cmd/microcloud/ask.go
+++ b/cmd/microcloud/ask.go
@@ -625,6 +625,19 @@ func (c *CmdControl) askRemotePool(systems map[string]InitSystem, autoSetup bool
 	return nil
 }
 
+func (c *CmdControl) askProceedIfNoOverlayNetwork() error {
+	proceedWithNoOverlayNetworking, err := c.asker.AskBool("FAN networking is not usable. Do you want to proceed with setting up an inoperable cluster? (yes/no) [default=no]: ", "no")
+	if err != nil {
+		return err
+	}
+
+	if proceedWithNoOverlayNetworking {
+		return nil
+	}
+
+	return fmt.Errorf("cluster bootstrapping aborted due to lack of usable networking")
+}
+
 func (c *CmdControl) askNetwork(sh *service.Handler, systems map[string]InitSystem, microCloudInternalSubnet *net.IPNet, autoSetup bool) error {
 	_, bootstrap := systems[sh.Name]
 	lxd := sh.Services[types.LXD].(*service.LXDService)
@@ -656,7 +669,7 @@ func (c *CmdControl) askNetwork(sh *service.Handler, systems map[string]InitSyst
 	// Automatic setup gets a basic fan setup.
 	if autoSetup {
 		if !fanUsable {
-			fmt.Println("FAN networking is not usable, skipping")
+			return c.askProceedIfNoOverlayNetwork()
 		}
 
 		return nil
@@ -665,7 +678,7 @@ func (c *CmdControl) askNetwork(sh *service.Handler, systems map[string]InitSyst
 	// Environments without OVN get a basic fan setup.
 	if sh.Services[types.MicroOVN] == nil {
 		if !fanUsable {
-			fmt.Println("FAN networking is not usable, skipping")
+			return c.askProceedIfNoOverlayNetwork()
 		}
 
 		return nil
@@ -787,7 +800,7 @@ func (c *CmdControl) askNetwork(sh *service.Handler, systems map[string]InitSyst
 	if !canOVN {
 		fmt.Println("No dedicated uplink interfaces detected, skipping distributed networking")
 		if !fanUsable {
-			fmt.Println("FAN networking is not usable, skipping")
+			return c.askProceedIfNoOverlayNetwork()
 		}
 
 		return nil

--- a/cmd/microcloud/ask.go
+++ b/cmd/microcloud/ask.go
@@ -628,29 +628,46 @@ func (c *CmdControl) askRemotePool(systems map[string]InitSystem, autoSetup bool
 func (c *CmdControl) askNetwork(sh *service.Handler, systems map[string]InitSystem, microCloudInternalSubnet *net.IPNet, autoSetup bool) error {
 	_, bootstrap := systems[sh.Name]
 	lxd := sh.Services[types.LXD].(*service.LXDService)
-	for peer, system := range systems {
-		if bootstrap {
-			system.TargetNetworks = []api.NetworksPost{lxd.DefaultPendingFanNetwork()}
-			if peer == sh.Name {
-				network, err := lxd.DefaultFanNetwork()
-				if err != nil {
-					return err
+
+	// Check if FAN networking is usable.
+	fanUsable, _, err := lxd.FanNetworkUsable()
+	if err != nil {
+		return err
+	}
+
+	if fanUsable {
+		for peer, system := range systems {
+			if bootstrap {
+				system.TargetNetworks = []api.NetworksPost{lxd.DefaultPendingFanNetwork()}
+				if peer == sh.Name {
+					network, err := lxd.DefaultFanNetwork()
+					if err != nil {
+						return err
+					}
+
+					system.Networks = []api.NetworksPost{network}
 				}
-
-				system.Networks = []api.NetworksPost{network}
 			}
-		}
 
-		systems[peer] = system
+			systems[peer] = system
+		}
 	}
 
 	// Automatic setup gets a basic fan setup.
 	if autoSetup {
+		if !fanUsable {
+			fmt.Println("FAN networking is not usable, skipping")
+		}
+
 		return nil
 	}
 
 	// Environments without OVN get a basic fan setup.
 	if sh.Services[types.MicroOVN] == nil {
+		if !fanUsable {
+			fmt.Println("FAN networking is not usable, skipping")
+		}
+
 		return nil
 	}
 
@@ -769,6 +786,10 @@ func (c *CmdControl) askNetwork(sh *service.Handler, systems map[string]InitSyst
 
 	if !canOVN {
 		fmt.Println("No dedicated uplink interfaces detected, skipping distributed networking")
+		if !fanUsable {
+			fmt.Println("FAN networking is not usable, skipping")
+		}
+
 		return nil
 	}
 

--- a/cmd/microcloud/main_init.go
+++ b/cmd/microcloud/main_init.go
@@ -200,14 +200,13 @@ func lookupPeers(s *service.Handler, autoSetup bool, iface *net.Interface, subne
 		}()
 	}
 
-	var timeAfter <-chan time.Time
+	timeoutDuration := time.Minute
 	if autoSetup {
-		timeAfter = time.After(5 * time.Second)
+		timeoutDuration = 5 * time.Second
 	}
 
-	if len(expectedSystems) > 0 {
-		timeAfter = time.After(1 * time.Minute)
-	}
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutDuration)
+	defer cancel()
 
 	expectedSystemsMap := make(map[string]bool, len(expectedSystems))
 	for _, system := range expectedSystems {
@@ -219,7 +218,7 @@ func lookupPeers(s *service.Handler, autoSetup bool, iface *net.Interface, subne
 	done := false
 	for !done {
 		select {
-		case <-timeAfter:
+		case <-ctx.Done():
 			done = true
 		case err := <-selectionCh:
 			if err != nil {
@@ -235,7 +234,7 @@ func lookupPeers(s *service.Handler, autoSetup bool, iface *net.Interface, subne
 				break
 			}
 
-			peers, err := mdns.LookupPeers(context.Background(), iface, mdns.Version, s.Name)
+			peers, err := mdns.LookupPeers(ctx, iface, mdns.Version, s.Name)
 			if err != nil {
 				return err
 			}

--- a/cmd/microcloud/main_init_preseed.go
+++ b/cmd/microcloud/main_init_preseed.go
@@ -498,8 +498,14 @@ func (p *Preseed) Parse(s *service.Handler, bootstrap bool) (map[string]InitSyst
 		systems[peer] = system
 	}
 
+	// Check if FAN networking is usable.
+	fanUsable, _, err := lxd.FanNetworkUsable()
+	if err != nil {
+		return nil, err
+	}
+
 	// Setup FAN network if OVN not available.
-	if len(ifaceByPeer) == 0 {
+	if len(ifaceByPeer) == 0 && fanUsable {
 		for peer, system := range systems {
 			if bootstrap {
 				system.TargetNetworks = append(system.TargetNetworks, lxd.DefaultPendingFanNetwork())

--- a/mdns/lookup.go
+++ b/mdns/lookup.go
@@ -148,7 +148,26 @@ func Lookup(ctx context.Context, iface *net.Interface, service string, size int)
 	params.Interface = iface
 	params.Entries = entriesCh
 	params.Timeout = 100 * time.Millisecond
-	err := mdns.Query(params)
+	ipv4Supported, ipv6Supported, err := checkIPStatus(iface.Name)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to check IP status: %w", err)
+	}
+
+	if !ipv4Supported {
+		logger.Info("IPv4 is not supported on this system network interface: disabling IPv4 mDNS", logger.Ctx{"iface": iface.Name})
+		params.DisableIPv4 = true
+	}
+
+	if !ipv6Supported {
+		logger.Info("IPv6 is not supported on this system network interface: disabling IPv6 mDNS", logger.Ctx{"iface": iface.Name})
+		params.DisableIPv6 = true
+	}
+
+	if params.DisableIPv4 && params.DisableIPv6 {
+		return nil, fmt.Errorf("No supported IP versions on the network interface %q", iface.Name)
+	}
+
+	err = mdns.Query(params)
 	if err != nil {
 		return nil, fmt.Errorf("Failed lookup: %w", err)
 	}

--- a/mdns/mdns.go
+++ b/mdns/mdns.go
@@ -48,3 +48,39 @@ func dnsTXTSlice(list []byte) []string {
 
 	return parts
 }
+
+// checkIPStatus checks if the interface is up, has multicast, and supports IPv4/IPv6.
+func checkIPStatus(iface string) (ipv4OK bool, ipv6OK bool, err error) {
+	netInterface, err := net.InterfaceByName(iface)
+	if err != nil {
+		return false, false, err
+	}
+
+	if netInterface.Flags&net.FlagUp != net.FlagUp {
+		return false, false, nil
+	}
+
+	if netInterface.Flags&net.FlagMulticast != net.FlagMulticast {
+		return false, false, nil
+	}
+
+	addrs, err := netInterface.Addrs()
+	if err != nil {
+		return false, false, err
+	}
+
+	for _, addr := range addrs {
+		ipNet, ok := addr.(*net.IPNet)
+		if !ok {
+			continue
+		}
+
+		if ipNet.IP.To4() != nil {
+			ipv4OK = true
+		} else if ipNet.IP.To16() != nil {
+			ipv6OK = true
+		}
+	}
+
+	return ipv4OK, ipv6OK, nil
+}

--- a/service/lxd.go
+++ b/service/lxd.go
@@ -1,13 +1,11 @@
 package service
 
 import (
-	"bufio"
 	"context"
 	"fmt"
 	"net"
 	"net/http"
 	"net/url"
-	"os"
 	"strings"
 	"time"
 
@@ -638,33 +636,14 @@ func (s *LXDService) waitReady(ctx context.Context, c lxd.InstanceServer, timeou
 }
 
 // defaultGatewaySubnetV4 returns subnet of default gateway interface.
-func defaultGatewaySubnetV4() (*net.IPNet, string, error) {
-	file, err := os.Open("/proc/net/route")
+func (s LXDService) defaultGatewaySubnetV4() (*net.IPNet, string, error) {
+	available, ifaceName, err := s.FanNetworkUsable()
 	if err != nil {
 		return nil, "", err
 	}
 
-	defer func() { _ = file.Close() }()
-
-	ifaceName := ""
-
-	scanner := bufio.NewReader(file)
-	for {
-		line, _, err := scanner.ReadLine()
-		if err != nil {
-			break
-		}
-
-		fields := strings.Fields(string(line))
-
-		if fields[1] == "00000000" && fields[7] == "00000000" {
-			ifaceName = fields[0]
-			break
-		}
-	}
-
-	if ifaceName == "" {
-		return nil, "", fmt.Errorf("No default gateway for IPv4")
+	if !available {
+		return nil, "", fmt.Errorf("No default IPv4 gateway available")
 	}
 
 	iface, err := net.InterfaceByName(ifaceName)

--- a/service/lxd_config.go
+++ b/service/lxd_config.go
@@ -1,9 +1,12 @@
 package service
 
 import (
+	"bufio"
 	"fmt"
 	"net"
+	"os"
 	"strconv"
+	"strings"
 
 	"github.com/canonical/lxd/shared/api"
 )
@@ -23,10 +26,42 @@ func (s LXDService) DefaultPendingFanNetwork() api.NetworksPost {
 	return api.NetworksPost{Name: "lxdfan0", Type: "bridge"}
 }
 
+// FanNetworkUsable checks if the current host is capable of using a Fan network.
+// It actually checks if there is a default IPv4 gateway available.
+func (s LXDService) FanNetworkUsable() (available bool, ifaceName string, err error) {
+	file, err := os.Open("/proc/net/route")
+	if err != nil {
+		return false, "", err
+	}
+
+	defer func() { _ = file.Close() }()
+
+	scanner := bufio.NewReader(file)
+	for {
+		line, _, err := scanner.ReadLine()
+		if err != nil {
+			break
+		}
+
+		fields := strings.Fields(string(line))
+
+		if fields[1] == "00000000" && fields[7] == "00000000" {
+			ifaceName = fields[0]
+			break
+		}
+	}
+
+	if ifaceName == "" {
+		return false, "", nil // There is no default gateway for IPv4
+	}
+
+	return true, ifaceName, nil
+}
+
 // DefaultFanNetwork returns the default Ubuntu Fan network configuration when
 // creating the finalized network.
 func (s LXDService) DefaultFanNetwork() (api.NetworksPost, error) {
-	underlay, _, err := defaultGatewaySubnetV4()
+	underlay, _, err := s.defaultGatewaySubnetV4()
 	if err != nil {
 		return api.NetworksPost{}, fmt.Errorf("Could not determine Fan overlay subnet: %w", err)
 	}

--- a/test/includes/microcloud.sh
+++ b/test/includes/microcloud.sh
@@ -5,7 +5,7 @@ unset_interactive_vars() {
   unset LOOKUP_IFACE LIMIT_SUBNET SKIP_SERVICE EXPECT_PEERS REUSE_EXISTING REUSE_EXISTING_COUNT \
     SETUP_ZFS ZFS_FILTER ZFS_WIPE \
     SETUP_CEPH CEPH_WARNING CEPH_FILTER CEPH_WIPE SETUP_CEPHFS CEPH_CLUSTER_NETWORK IGNORE_CEPH_NETWORKING \
-    SETUP_OVN OVN_WARNING OVN_FILTER IPV4_SUBNET IPV4_START IPV4_END DNS_ADDRESSES IPV6_SUBNET
+    PROCEED_WITH_NO_OVERLAY_NETWORKING SETUP_OVN OVN_WARNING OVN_FILTER IPV4_SUBNET IPV4_START IPV4_END DNS_ADDRESSES IPV6_SUBNET
 }
 
 # microcloud_interactive: outputs text that can be passed to `TEST_CONSOLE=1 microcloud init`
@@ -19,24 +19,25 @@ microcloud_interactive() {
   EXPECT_PEERS=${EXPECT_PEERS:-}                 # wait for this number of systems to be available to join the cluster.
   REUSE_EXISTING=${REUSE_EXISTING:-}              # (yes/no) incorporate an existing clustered service.
   REUSE_EXISTING_COUNT=${REUSE_EXISTING_COUNT:-0} # (number) number of existing clusters to incorporate.
-  SETUP_ZFS=${SETUP_ZFS:-}                       # (yes/no) input for initiating ZFS storage pool setup.
-  ZFS_FILTER=${ZFS_FILTER:-}                     # filter string for ZFS disks.
-  ZFS_WIPE=${ZFS_WIPE:-}                         # (yes/no) to wipe all disks.
-  SETUP_CEPH=${SETUP_CEPH:-}                     # (yes/no) input for initiating CEPH storage pool setup.
-  SETUP_CEPHFS=${SETUP_CEPHFS:-}                 # (yes/no) input for initialising CephFS storage pool setup.
-  CEPH_WARNING=${CEPH_WARNING:-}                 # (yes/no) input for warning about eligible disk detection.
-  CEPH_FILTER=${CEPH_FILTER:-}                   # filter string for CEPH disks.
-  CEPH_WIPE=${CEPH_WIPE:-}                       # (yes/no) to wipe all disks.
+  SETUP_ZFS=${SETUP_ZFS:-}                        # (yes/no) input for initiating ZFS storage pool setup.
+  ZFS_FILTER=${ZFS_FILTER:-}                      # filter string for ZFS disks.
+  ZFS_WIPE=${ZFS_WIPE:-}                          # (yes/no) to wipe all disks.
+  SETUP_CEPH=${SETUP_CEPH:-}                      # (yes/no) input for initiating CEPH storage pool setup.
+  SETUP_CEPHFS=${SETUP_CEPHFS:-}                  # (yes/no) input for initialising CephFS storage pool setup.
+  CEPH_WARNING=${CEPH_WARNING:-}                  # (yes/no) input for warning about eligible disk detection.
+  CEPH_FILTER=${CEPH_FILTER:-}                    # filter string for CEPH disks.
+  CEPH_WIPE=${CEPH_WIPE:-}                        # (yes/no) to wipe all disks.
   CEPH_CLUSTER_NETWORK=${CEPH_CLUSTER_NETWORK:-} # (default: MicroCloud internal subnet or Ceph public network if specified previously) input for setting up a cluster network.
   IGNORE_CEPH_NETWORKING=${IGNORE_CEPH_NETWORKING:-} # (yes/no) input for ignoring Ceph network setup. Set it to `yes` during `microcloud add` .
-  SETUP_OVN=${SETUP_OVN:-}                       # (yes/no) input for initiating OVN network setup.
-  OVN_WARNING=${OVN_WARNING:-}                   # (yes/no) input for warning about eligible interface detection.
-  OVN_FILTER=${OVN_FILTER:-}                     # filter string for OVN interfaces.
-  IPV4_SUBNET=${IPV4_SUBNET:-}                   # OVN ipv4 gateway subnet.
-  IPV4_START=${IPV4_START:-}                     # OVN ipv4 range start.
-  IPV4_END=${IPV4_END:-}                         # OVN ipv4 range end.
-  DNS_ADDRESSES=${DNS_ADDRESSES:-}               # OVN custom DNS addresses.
-  IPV6_SUBNET=${IPV6_SUBNET:-}                   # OVN ipv6 range.
+  PROCEED_WITH_NO_OVERLAY_NETWORKING=${PROCEED_WITH_NO_OVERLAY_NETWORKING:-} # (yes/no) input for proceeding without overlay networking.
+  SETUP_OVN=${SETUP_OVN:-}                        # (yes/no) input for initiating OVN network setup.
+  OVN_WARNING=${OVN_WARNING:-}                    # (yes/no) input for warning about eligible interface detection.
+  OVN_FILTER=${OVN_FILTER:-}                      # filter string for OVN interfaces.
+  IPV4_SUBNET=${IPV4_SUBNET:-}                    # OVN ipv4 gateway subnet.
+  IPV4_START=${IPV4_START:-}                      # OVN ipv4 range start.
+  IPV4_END=${IPV4_END:-}                          # OVN ipv4 range end.
+  DNS_ADDRESSES=${DNS_ADDRESSES:-}                # OVN custom DNS addresses.
+  IPV6_SUBNET=${IPV6_SUBNET:-}                    # OVN ipv6 range.
 
   setup="
 ${LOOKUP_IFACE}                                         # filter the lookup interface
@@ -84,6 +85,13 @@ $([ "${SETUP_CEPH}" = "yes" ] && printf -- "---")
 $([ "${CEPH_WIPE}"  = "yes" ] && printf "select-all")   # wipe all disks
 $([ "${SETUP_CEPH}" = "yes" ] && printf -- "---")
 ${SETUP_CEPHFS}
+$(true)                                                 # workaround for set -e
+"
+fi
+
+if [ -n "${PROCEED_WITH_NO_OVERLAY_NETWORKING}" ]; then
+  setup="${setup}
+${PROCEED_WITH_NO_OVERLAY_NETWORKING}                   # agree to proceed without overlay networking (neither FAN nor OVN networking) (yes/no)
 $(true)                                                 # workaround for set -e
 "
 fi

--- a/test/suites/basic.sh
+++ b/test/suites/basic.sh
@@ -47,6 +47,46 @@ test_interactive() {
     validate_system_lxd "${m}" 3 disk1
   done
 
+  # Reset the systems with just LXD and no IPv6 support.
+  reset_systems 3 3 1
+
+  for m in micro01 micro02 micro03 ; do
+    lxc exec "${m}" -- echo 1 > /proc/sys/net/ipv6/conf/all/disable_ipv6
+    lxc exec "${m}" -- snap disable microceph || true
+    lxc exec "${m}" -- snap disable microovn || true
+    lxc exec "${m}" -- snap restart microcloud
+  done
+
+  echo "Creating a MicroCloud with ZFS storage and no IPv6 support"
+  microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init > out"
+
+  lxc exec micro01 -- tail -1 out | grep "MicroCloud is ready" -q
+  for m in micro01 micro02 micro03 ; do
+    validate_system_lxd "${m}" 3 disk1
+  done
+
+  # Reset the systems with just LXD and no IPv4 support.
+  gw_net_addr=$(lxc network get lxdbr0 ipv4.address)
+  lxc network set lxdbr0 ipv4.address none
+  reset_systems 3 3 1
+
+  for m in micro01 micro02 micro03 ; do
+    lxc exec "${m}" -- snap disable microceph || true
+    lxc exec "${m}" -- snap disable microovn || true
+    lxc exec "${m}" -- snap restart microcloud
+  done
+
+  export PROCEED_WITH_NO_OVERLAY_NETWORKING="no" # This will avoid to setup the cluster if no overlay networking is available.
+  echo "Creating a MicroCloud with ZFS storage and no IPv4 support"
+  ! microcloud_interactive | lxc exec micro01 -- sh -c "microcloud init 2> err" || false
+
+  # Ensure we error out due to a lack of usable overlay networking.
+  lxc exec micro01 -- cat err | grep "cluster bootstrapping aborted due to lack of usable networking" -q
+
+  # Set the IPv4 address back to the original value.
+  lxc network set lxdbr0 ipv4.address "${gw_net_addr}"
+  unset PROCEED_WITH_NO_OVERLAY_NETWORKING
+
   # Reset the systems and install microceph.
   reset_systems 3 3 1
 

